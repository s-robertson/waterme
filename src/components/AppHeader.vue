<template>
  <div class="app-header">
    <div class="app-header__container">
      <router-link to="/" class="app-header__logo-container">
        <PlantIcon class="app-header__logo" alt="" role="presentation" />
        <span class="app-header__logo-text">Water Me</span>
      </router-link>
      <div class="app-header__menu">
        <ButtonLogout />
      </div>
    </div>
  </div>
</template>

<script>
import PlantIcon from "@/assets/plant.svg";
import ButtonLogout from "@/components/ButtonLogout";

export default {
  name: "AppHeader",
  components: {
    ButtonLogout,
    PlantIcon
  }
};
</script>

<style lang="scss">
@import "../assets/scss/variables";

.app-header {
  background: $brand-secondary;
  padding: 1rem 0;
  margin-bottom: 1.5rem;
  font-size: 1.5rem;
}

.app-header__container {
  max-width: $app-max-width;
  margin: 0 auto;
  padding: 0 1.5rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.app-header__logo-container {
  text-decoration: none;
}

.app-header__logo {
  width: 50px;
  height: 50px;
  fill: #00dcff;
  margin-right: 0.5rem;
  vertical-align: middle;
}

.app-header__logo-text {
  display: inline-block;
  padding-top: 0.5rem;
  vertical-align: middle;
}
</style>
