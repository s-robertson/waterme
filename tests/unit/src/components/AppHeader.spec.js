import { createLocalVue, shallowMount } from "@vue/test-utils";
import VueRouter from "vue-router";
import AppHeader from "@/components/AppHeader";

const localVue = createLocalVue();

localVue.use(VueRouter);

describe("AppHeader component", () => {
  it("Should render correctly", () => {
    const wrapper = shallowMount(AppHeader, { localVue });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
