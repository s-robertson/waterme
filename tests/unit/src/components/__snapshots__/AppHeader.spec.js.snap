// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader component Should render correctly 1`] = `
<div class="c-app-header">
  <div class="c-app-header__container">
    <router-link to="/" class="c-app-header__logo-container">
      <planticon-stub alt="" role="presentation" class="c-app-header__logo"></planticon-stub> <span class="c-app-header__logo-text">Water Me</span>
    </router-link>
    <div class="c-app-header__menu">
      <buttonlogout-stub></buttonlogout-stub>
    </div>
  </div>
</div>
`;
